src/docfx/lib/git/GitUtility.cs

```diff
@@ -213,6 +213,47 @@ public static void CheckMergeConflictMarker(string content, string file)
             }
         }
 
+        public static unsafe bool TryGetContentFromHistory(string repoPath, string filePath, string committish, out string content)
+        {
+            content = null;
+            if (git_repository_open(out var repo, repoPath) != 0)
+            {
+                return false;
+            }
+
+            if (git_revparse_single(out var commit, repo, committish) != 0)
+            {
+                git_repository_free(repo);
+                return false;
+            }
+
+            if (git_commit_tree(out var tree, commit) != 0)
+            {
+                git_repository_free(repo);
+                return false;
+            }
+
+            if (git_tree_entry_bypath(out var entry, tree, filePath) != 0)
+            {
+                git_repository_free(repo);
+                return false;
+            }
+
+            var obj = git_tree_entry_id(entry);
+            if (git_blob_lookup(out var blob, repo, obj) != 0)
+            {
+                git_tree_entry_free(entry);
+                git_repository_free(repo);
+                return false;
+            }
+
+            content = git_blob_rawcontent(blob);
```

Comment:
This line will fail if the blob uses a different encoding or memory layout than `System.String`.

Read the string from the memory chunk defined by `git_blob_rawcontent` and `git_blob_rawsize`. `git_blob_rawcontent` should return `IntPtr` instead of `string`.

Reply:
ok, thanks for this, try to change it....

---
In reply to: [236448956](https://github.com/dotnet/docfx/pull/3677#discussion_r236448956) [](ancestors = 236448956)